actors/evm/src/interpreter/instructions/call.rs

```diff
@@ -250,11 +248,10 @@ pub fn call_generic<RT: Runtime>(
                         };
                         // TODO: support IPLD codecs #758
                         let params = RawBytes::serialize(BytesSer(input_data))?;
-                        if kind == CallKind::StaticCall {
-                            system.send_read_only(&dst_addr, method, params)
-                        } else {
-                            system.send(&dst_addr, method, params, TokenAmount::from(&value))
-                        }
+                        let value = TokenAmount::from(&value);
+                        let gas_limit = if !gas.is_zero() { Some(gas.as_u64()) } else { None };
+                        let read_only = system.readonly || kind == CallKind::StaticCall;
```

Comment:
We don't need to propagate our own read-only value (and probably shouldn't). Let the system handle that.

Reply:
done.